plugins/providers/maxcompute/provider.go

```diff
@@ -86,34 +86,28 @@ func (p *provider) CreateConfig(pc *domain.ProviderConfig) error {
 
 func (p *provider) GetResources(ctx context.Context, pc *domain.ProviderConfig) ([]*domain.Resource, error) {
 	var resources = make([]*domain.Resource, 0)
-
 	var availableResourceTypes = pc.GetResourceTypes()
-	var retrieveSchemas = slices.Contains(availableResourceTypes, resourceTypeSchema) || slices.Contains(availableResourceTypes, resourceTypeTable)
-	var retrieveProject = retrieveSchemas || slices.Contains(availableResourceTypes, resourceTypeProject)
-
-	var project *domain.Resource
-	var accountId string
-	var err error
-	if retrieveProject {
-		project, accountId, err = p.getProject(ctx, pc)
-		if err != nil {
-			return nil, err
-		}
+	if !slices.GenericsSliceContainsOne(availableResourceTypes, resourceTypeProject, resourceTypeSchema, resourceTypeTable) {
+		return resources, nil
+	}
+
+	project, accountId, err := p.getProject(ctx, pc)
+	if err != nil {
+		return nil, err
 	}
 
-	var schemas = make([]*domain.Resource, 0)
-	if retrieveSchemas {
-		schemas, err = p.getSchemasFromProject(ctx, pc, "", accountId, project)
+	if slices.GenericsSliceContainsOne(availableResourceTypes, resourceTypeSchema, resourceTypeTable) {
+		project.Children, err = p.getSchemasFromProject(ctx, pc, "", accountId, project)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	if slices.Contains(availableResourceTypes, resourceTypeTable) {
+	if slices.GenericsSliceContainsOne(availableResourceTypes, resourceTypeTable) {
```

Comment:
similar, `slices.Contains(availableResourceTypes, resourceTypeTable)` (from "slices" go util package) already does the same